cmd/operator-sdk/new/cmd.go

```diff
@@ -180,26 +178,16 @@ func doGoScaffold() error {
 		s.BoilerplatePath = headerFile
 	}
 
-	var err error
-	switch m := projutil.DepManagerType(depManager); m {
-	case projutil.DepManagerDep:
-		err = s.Execute(cfg, &scaffold.GopkgToml{})
-	case projutil.DepManagerGoMod:
-		if goModOn, merr := projutil.GoModOn(); merr != nil {
-			return merr
-		} else if !goModOn {
-			return errors.New(`dependency manager "modules" requires working directory to be in $GOPATH/src` +
-				` and GO111MODULE=on, or outside of $GOPATH/src and GO111MODULE="on", "auto", or unset. More info: https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md#go-modules`)
-		}
-		err = s.Execute(cfg, &scaffold.GoMod{}, &scaffold.Tools{})
-	default:
-		err = projutil.ErrNoDepManager
-	}
-	if err != nil {
-		return fmt.Errorf("dependency manager file scaffold failed: (%v)", err)
+	if goModOn, merr := projutil.GoModOn(); merr != nil {
+		return merr
+	} else if !goModOn {
+		return errors.New(`using go modules requires GO111MODULE="on", "auto", or unset.` +
+			` More info: https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md#go-modules`)
```

Comment:
This looks the same as `checkGoModules`. I'm okay with duplicating code in order to make it more readable since maintenance is done way more than new features. Not familiar enough to know if this could be reused here though.

Reply:
Yeah, I think it makes sense to move `checkGoModules` to `projutil` to make it reusable. Good catch.